### PR TITLE
Fix LB-76

### PR DESCRIPTION
**Pull request by mayhem**
Ensure that only one max_ts or min_ts can be used, not both.
Make both exclusive.

### CODE REVIEW

**Comment by gentlecat** [comment]
Looks good to me. I guess docs and pagination will need to be updated. :sheep: :honeybee::honeybee::honeybee:

File: listenstore/listenstore/listenstore.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def fetch_listens_for_range(self, uid, date_range, from_id, to_id, limit=None, o
         """
         query = """SELECT * FROM listens WHERE uid = %(uid)s AND """ + \
                 range_keys(len(date_range)) + \
-                """ AND id >= %(from_id)s AND id <= %(to_id)s
+                """ AND id > %(from_id)s AND id < %(to_id)s
```
Comment (alastair) [inline listenstore/listenstore/listenstore.py]: This doesn't validate that min_ts/max_ts are exclusive. Does this matter, with the checks in the view?

File: webserver/views/api.py [hunk 2]
```diff
@@ -75,12 +75,19 @@ def submit_listen():
...
+        _log_and_raise_400("You may only specify max_ts or min_ts, not both.")
+
     cassandra = webserver.create_cassandra()
     listens = cassandra.fetch_listens(
         user_id,
         limit=min(_parse_int_arg("count", DEFAULT_ITEMS_PER_GET), MAX_ITEMS_PER_GET),
-        from_id=_parse_int_arg("max_ts"),
-        to_id=_parse_int_arg("min_ts"),
+        from_id=min_ts,
+        to_id=max_ts,
```
Comment (alastair) [inline webserver/views/api.py]: does listenstore accept None as an argument? what's it's behaviour?

File: listenstore/listenstore/listenstore.py (hunk 1 above)
Comment (alastair) [inline listenstore/listenstore/listenstore.py]: in our user profile view, we call `listenstore.fetch_listens`, so we should have the "one of min or max" check there too

Comment (alastair) [inline listenstore/listenstore/listenstore.py]: we can choose asc/desc as the order based on if we get passed a max ts or min ts

File: webserver/views/api.py (hunk 2 above)
Comment (alastair) [inline webserver/views/api.py]: after looking at the code for `fetch_listens`, None will be converted into a upper or lower bound (beginning of last.fm, or 1 hour in the future from now). The count is used to quit

**Comment by alastair** [comment]
:rocket:

**Pull request merged**